From fa87940e8750c74c0f593b7db35d30d78b82b654 Mon Sep 17 00:00:00 2001
From: Hien Dang <hien.dang.eb@renesas.com>
Date: Wed, 17 Aug 2016 15:12:59 +0700
Subject: [PATCH 1668/2066] serial: sh-sci: Use pm_runtime_get/put when
 suspend

Due to the commit '39dd0f234fc37d ("PM / Domains: Allow runtime PM during
system PM phases")', runtime PM status might be suspended when back up
module registers. Therefore, it is necessary to ensure device is
on during suspend by using pm_runtime_get/put.

Signed-off-by: Hien Dang <hien.dang.eb@renesas.com>
[zou:Original patch taken from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git
 v4.9/rcar-3.5.1]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/tty/serial/sh-sci.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/tty/serial/sh-sci.c b/drivers/tty/serial/sh-sci.c
index 379b76a..5b817b4 100644
--- a/drivers/tty/serial/sh-sci.c
+++ b/drivers/tty/serial/sh-sci.c
@@ -3367,10 +3367,12 @@ static __maybe_unused int sci_suspend(struct device *dev)
 	int ret = 0;
 
 	if (sport) {
+		pm_runtime_get_sync(sport->port.dev);
 		uart_suspend_port(&sci_uart_driver, &sport->port);
 #ifdef CONFIG_RCAR_DDR_BACKUP
 		ret = sci_save_regs(dev);
 #endif /* CONFIG_RCAR_DDR_BACKUP */
+		pm_runtime_put(sport->port.dev);
 	}
 
 	return ret;
-- 
1.7.5.4

